bigquery_etl/shredder/delete.py

```diff
@@ -207,18 +218,16 @@ def create_job(client):
         )
         query = reformat(
             f"""
-            {"SELECT * FROM" if dry_run and read_only else "DELETE"}
+            {"DELETE" if use_dml else "SELECT * FROM"}
               `{sql_table_id(target)}`
             WHERE
-              ({field_condition})
-              AND {partition_condition}
+              {"" if use_dml else "NOT "}({field_condition})
```

Comment:
Is it possible the field condition could evaluate to null?  Because simply applying `NOT` won't select the exact inverse if nulls are involved.

If nulls are involved you could do something like
```suggestion
              COALESCE(({field_condition}), FALSE) = {"TRUE" if use_dml else "FALSE"}
```

Reply:
nulls are not a concern here.

Reply:
OK, so it looks like the assumption is the deletion requests (or similar events) used as sources never have null values in the relevant ID columns.

Personally I'd still be inclined to code defensively against possible nulls, as the consequence of a possible null value there would be to accidentally delete all data in the partition (`IN` returns null if there's no positive match in the search set and the search set contains any nulls).

Reply:
Okay, then how about:
```suggestion
             ({field_condition}) {"" if use_dml else " IS NOT TRUE"}
```